### PR TITLE
[jit] Cleanup documentation around `script` and `trace`

### PR DESCRIPTION
**Pull request by driazati**
Stacked PRs
 * #24258 - [jit] Add `trace_module` to docs
 * **#24208 - [jit] Cleanup documentation around `script` and `trace`**

Examples / info was duplicated between `ScriptModule`, `script`, and
`trace`, so this PR consolidates it and moves some things around to make
the docs more clear.

For reviewers, if you want to see the rendered output, ping me for a
link

Differential Revision: [D16746236](https://our.internmc.facebook.com/intern/diff/16746236/)

### CODE REVIEW

File: torch/jit/__init__.py [hunk 1]
```diff
@@ -688,12 +688,32 @@ def trace(func,
     Trace a function and return an executable ``ScriptModule`` or ``torch._C.Function``
     that will be optimized using just-in-time compilation.
 
-    .. warning::
+    Using ``torch.jit.trace`` and :func:`torch.jit.trace_module<torch.jit.trace_module>`, you can turn an existing module or Python
+    function into a TorchScript ``torch._C.Function`` or ``ScriptModule``. You must provide example inputs,
```
Comment (zdevito) [inline torch/jit/__init__.py]: Please do not document _C as a public facing API. We should make the functions torch.jit.Function

**Review by zdevito — APPROVED** [review]

File: torch/jit/__init__.py [hunk 2]
```diff
@@ -996,10 +1041,17 @@ def script(obj, optimize=None, _frames_up=0, _rcb=None):
     r"""
     Scripting a function or ``nn.Module`` will inspect the source code, compile
     it as TorchScript code using the TorchScript compiler, and return a ``ScriptModule`` or
-    ``torch._C.Function``.
+    ``torch._C.Function``. TorchScript itself is a subset of the Python language, so not all
+    features in Python work, but we provide enough functionality to compute on
+    tensors and do control-dependent operations. For a complete guide, see the
+    `TorchScript Language Reference`_.
```
Comment (ZolotukhinM) [inline torch/jit/__init__.py]: Redundant underscore in the end?

**Review by ZolotukhinM — APPROVED** [review]
> For reviewers, if you want to see the rendered output, ping me for a
link

You can click "View file" in the diff view for rendered version btw.

File: torch/jit/__init__.py (hunk 2 above)
Comment (driazati) [reply torch/jit/__init__.py]: The `View File` renders it as markdown, a trailing underscore in `rST` is a link (this one is automatically created from the header with this same name

Comment (Krovatkin) [reply torch/jit/__init__.py]: I saw `_` appended in other spots and I was always wondering what it was used for> 😄

**Pull request closed without merging**